build-system/tasks/integration.js

```diff
@@ -0,0 +1,201 @@
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'using strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const babelify = require('babelify');
+const defaultConfig = require('./karma.conf');
+const log = require('fancy-log');
+const testConfig = require('../config');
+const {
+  createKarmaServer,
+  getAdTypes,
+  getBrowserConfig,
+  maybePrintArgvMessages,
+  maybeSetCoverageConfig,
+  refreshKarmaWdCache,
+  runTestInBatches,
+  startTestServer,
+} = require('./runtime-test/helpers');
+const {clean} = require('./clean');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {dist} = require('./dist');
+const {green, yellow, cyan, red} = require('ansi-colors');
+const {isTravisBuild} = require('../travis');
+
+function shouldNotRun() {
+  if (argv.saucelabs) {
+    if (!process.env.SAUCE_USERNAME) {
+      throw new Error('Missing SAUCE_USERNAME Env variable');
+    }
+    if (!process.env.SAUCE_ACCESS_KEY) {
+      throw new Error('Missing SAUCE_ACCESS_KEY Env variable');
+    }
+  }
+
+  return false;
+}
+
+async function maybeBuild() {
+  if (argv.nobuild) {
+    return;
+  }
+  argv.fortesting = true;
+  argv.compiled = true;
+  await clean();
+  await dist();
+}
+
+function setConfigOverrides(config) {
+  config.singleRun = !argv.watch && !argv.w;
+  config.client.mocha.grep = !!argv.grep;
+  config.client.verboseLogging = !!argv.verbose || !!argv.v;
+  config.client.captureConsole = !!argv.verbose || !!argv.v || !!argv.files;
+
+  config.browserify.configure = function(bundle) {
+    bundle.on('prebundle', function() {
+      log(green('Transforming tests with'), cyan('browserify') + green('...'));
+    });
+    bundle.on('transform', function(tr) {
+      if (tr instanceof babelify) {
+        tr.once('babelify', function() {
+          process.stdout.write('.');
+        });
+      }
+    });
+  };
+
+  // c.client is available in test browser via window.parent.karma.config
+  config.client.amp = {
+    useCompiledJs: !!argv.compiled,
+    saucelabs: !!argv.saucelabs,
+    singlePass: !!argv.single_pass,
+    adTypes: getAdTypes(),
+    mochaTimeout: defaultConfig.client.mocha.timeout,
+    propertiesObfuscated: !!argv.single_pass,
+    testServerPort: defaultConfig.client.testServerPort,
+    testOnIe: config.browsers.includes('IE'),
+  };
+}
+
+function getFileConfig() {
+  const files = testConfig.commonIntegrationTestPaths;
+
+  if (argv.files) {
+    return {'files': files.concat(argv.files)};
+  }
+
+  return {'files': files.concat(testConfig.integrationTestPaths)};
+}
+
+function getReporterConfig() {
+  let {reporters} = defaultConfig;
+
+  if ((!isTravisBuild() && argv.testnames) || (argv.files && !argv.saucelabs)) {
```

Comment:
This can be simplified to...
```js
if ((argv.testnames || argv.files) && !argv.saucelabs)
```
...without `&& !argv.saucelabs` if the `mocha` and `saucelabs` reporters can be used together.